### PR TITLE
Ignore WMS extent

### LINKED ISSUE #4201: Suppress extent limit for WMS layers
# Feature Request

**Is your feature request related to a problem? Please describe.**
To mitigate two issues:

1.  a layer with symbology, when WMS ignores the size of the symbols and does not take them into account when calculating the extent. Hence the symbols are cropped on the edges.
2. for layers added from Layman which use external URI to PostGIS database, the update of extent is not triggered after edit. Thus, when a user is adding features outside of current extent, this is not reflected and the features are invisible (because they are not requested by OpenLayers)

**Describe the solution you'd like**
Add a checkbox to LayerManager saying "ignore layer extent" which will set the extent to undefined and force the layer to request GetMap for any part of the world. Disabled by default. Enabled by default for Layman/PostGIS layers.

**Describe alternatives you've considered**
Make the extent "a bit" larger for layers with symbology. This is a pure guessing however...
Instead of checkbox, implement slider saying "enlarge layer extent by N units"

**Additional context**
https://redmine.ccss.cz/issues/9578

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description

Add 'Igonre extent' toggle for WMS layers.

## Related issues or pull requests

closes #4201 

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Review by jmacura — CHANGES_REQUESTED** [review]
Works great for layers added via URL. Does not work for layers added via HsConfig.

Also, I would change "extentStash" to "originalExtent" or "sourceExtent" or "wmsExtent" or "extentFromWms" or something alike..

File: projects/hslayers/src/common/layer-extensions.ts [hunk 1]
```diff
@@ -56,6 +57,7 @@ const VIRTUAL_ATTRIBUTES = 'virtualAttributes';
 const WFS_URL = 'wfsUrl';
 const WORKSPACE = 'workspace';
 export const DOM_FEATURE_LINKS = 'domFeatureLinks';
+const WMS_EXTENT_STASH = 'wmsExtentStash';
```
[the remark this answers is not in the document]
Comment (FilipLeitner) [reply projects/hslayers/src/common/layer-extensions.ts:60]: Indeed

File: projects/hslayers/src/common/get-capabilities/wms-get-capabilities-response.interface.ts [hunk 2]
```diff
@@ -114,7 +114,15 @@ export type WmsDimension = {
   current?: boolean;
   values: string | string[];
 };
-export type WmsLayer = {
+
+/***
+ * HsLayers extension of default WMS Layer object
+ */
+export type HsLayersWmsLayer = {
```
Comment (jmacura) [inline projects/hslayers/src/common/get-capabilities/wms-get-capabilities-response.interface.ts]: I would defined the types the other way around...

**Review by jmacura — APPROVED** [review]
See last commit (rename of types).

Seems to work for init layers as well now 👍

File: projects/hslayers/src/common/get-capabilities/wms-get-capabilities-response.interface.ts (hunk 2 above)
Comment (FilipLeitner) [reply projects/hslayers/src/common/get-capabilities/wms-get-capabilities-response.interface.ts]: I was thinking about it as a hslayers extension to the base response, but yes this makes more sense. Thanks for the change

**Pull request merged**